### PR TITLE
clock_control: nordic: move to new DT API

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Use the new devicetree API to access dts data.

### CODE REVIEW

File: drivers/clock_control/nrf_clock_calibration.c [hunk 1]
```diff
@@ -215,6 +212,15 @@ static void measure_temperature(struct k_work *work)
 			started ? "started" : "skipped", diff);
 }
 
+#if DT_HAS_NODE(DT_NODELABEL(temp))
+static inline struct device *temp_device(void)
+{
+	return device_get_binding(DT_LABEL(DT_NODELABEL(temp)));
```
Comment (galak) [inline drivers/clock_control/nrf_clock_calibration.c:218]: Does the `temp` device need to be NODELABEL based?  Seems like DT_INST would be fine in this case.  We should only use NODELABEL in the cases that its really required.

**Review by galak — CHANGES_REQUESTED** [review]
Change 'temp' from DT_NODELABEL to DT_INST.

File: drivers/clock_control/nrf_clock_calibration.c (hunk 1 above)
Comment (mbolivar-nordic) [reply drivers/clock_control/nrf_clock_calibration.c:218]: > Does the `temp` device need to be NODELABEL based?

Yes, that is my interpretation: https://github.com/zephyrproject-rtos/zephyr/pull/23744#discussion_r397745301

Comment (galak) [reply drivers/clock_control/nrf_clock_calibration.c:218]: I see 2 ways a device should or shouldn't exist:
1. There is no node w/compat for the device in the dt
2. there is a node w/compat and `status = disabled`

Comment (galak) [reply drivers/clock_control/nrf_clock_calibration.c:218]: In the thread @anangl even gives the snippet using `DT_INST` form.

**Comment by mbolivar-nordic** [comment]
> Change 'temp' from DT_NODELABEL to DT_INST.

doen

**Review by galak — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**